src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1684,16 +1783,23 @@ static int add_object_vtable_internal(
                 goto fail;
         }
 
```

Comment:
in the code, I verify the size of the table every time it is needed. A more robust alternative would be to copy the table, and so to upgrade the old version(s) all in one place, so that there is no risk of mishandling.
But this too comes with a cost: dynamic allocation, more memory, and also need to free it.